django_admin_lightweight_date_hierarchy/utils.py

```diff
@@ -0,0 +1,44 @@
+import datetime
+import calendar
+
+def get_date_hierarchy_drilldown_on_past_dates(year_lookup, month_lookup, asof, past_years=3):
+    """Drill-down only on past dates.
+
+    year_lookup (int or None)
+    month_lookup (int or None)
+    asof (datetime.datetime aware):
+        Current date.
+    past_years (int):
+        Number of past years to return for year drilldown.
+
+    Yields (datetime.date):
+        List of dates to drill down on.
+    """
+    today = asof.date()
+
+    if year_lookup is None and month_lookup is None:
+        # Past 3 years.
+        return (
+            datetime.date(y, 1, 1)
+            for y in range(today.year - past_years + 1, today.year + 1)
+        )
+
+    elif year_lookup is not None and month_lookup is None:
+        # Past months of selected year.
+        this_month = today.replace(day=1)
+        return (
+            month for month in (
```

Comment:
`month` is shadowed here and for two different things: the month number and the month `date`. Should use different names since it's confusing.

Same below for `day`.

Reply:
Are you shure it quilify as shadow if they operate in different scopes?

Reply:
Yes, shadow by definition is different scope. But that doesn't matter, point is using the same name is confusing.